ext/afform/core/Civi/Api4/Action/Afform/Submit.php

```diff
@@ -204,7 +205,7 @@ private static function getRequiredFieldError(string $apiEntity, string $fieldNa
     $fullDefn = FormDataModel::getField($apiEntity, $fieldName, 'create');
     $isRequired = $attributes['defn']['required'] ?? $fullDefn['required'] ?? FALSE;
     if ($isRequired) {
-      $label = $attributes['defn']['label'] ?? $fullDefn['label'];
+      $label = $attributes['defn']['label'] ?? $fullDefn['label'] ?? $fieldName;
```

Comment:
Sometimes label is not set eg. for a select2 and you get the unhelpful error message " is not a required field". This falls back to fieldName so we can work it out.